From a04fe4b347e563d7ff8d116fde6c6f26646ffadd Mon Sep 17 00:00:00 2001
From: Hushan Jia <hushan.jia@gmail.com>
Date: Sun, 21 Feb 2016 21:10:24 -0800
Subject: [PATCH 102/116] ipcs: show gid instead of uid

ipcs -i incorrectly used uid where it should be gid

Upstream: https://github.com/karelzak/util-linux/commit/e0bbe3d6ffed13ca5e4af450b8de775ddc5b30f3
Addresses: https://bugzilla.redhat.com/show_bug.cgi?id=1358095
Signed-off-by: Hushan Jia <hushan.jia@gmail.com>
---
 sys-utils/ipcs.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/sys-utils/ipcs.c b/sys-utils/ipcs.c
index 1843cd5..1db7fd0 100644
--- a/sys-utils/ipcs.c
+++ b/sys-utils/ipcs.c
@@ -564,7 +564,7 @@ static void print_shm(int shmid, int unit)
 
 	printf(_("\nShared memory Segment shmid=%d\n"), shmid);
 	printf(_("uid=%u\tgid=%u\tcuid=%u\tcgid=%u\n"),
-	       shmdata->shm_perm.uid, shmdata->shm_perm.uid,
+	       shmdata->shm_perm.uid, shmdata->shm_perm.gid,
 	       shmdata->shm_perm.cuid, shmdata->shm_perm.cgid);
 	printf(_("mode=%#o\taccess_perms=%#o\n"), shmdata->shm_perm.mode,
 	       shmdata->shm_perm.mode & 0777);
@@ -594,7 +594,7 @@ void print_msg(int msgid, int unit)
 
 	printf(_("\nMessage Queue msqid=%d\n"), msgid);
 	printf(_("uid=%u\tgid=%u\tcuid=%u\tcgid=%u\tmode=%#o\n"),
-	       msgdata->msg_perm.uid, msgdata->msg_perm.uid,
+	       msgdata->msg_perm.uid, msgdata->msg_perm.gid,
 	       msgdata->msg_perm.cuid, msgdata->msg_perm.cgid,
 	       msgdata->msg_perm.mode);
 	ipc_print_size(unit, unit == IPC_UNIT_HUMAN ? _("csize=") : _("cbytes="),
@@ -627,7 +627,7 @@ static void print_sem(int semid)
 
 	printf(_("\nSemaphore Array semid=%d\n"), semid);
 	printf(_("uid=%u\t gid=%u\t cuid=%u\t cgid=%u\n"),
-	       semdata->sem_perm.uid, semdata->sem_perm.uid,
+	       semdata->sem_perm.uid, semdata->sem_perm.gid,
 	       semdata->sem_perm.cuid, semdata->sem_perm.cgid);
 	printf(_("mode=%#o, access_perms=%#o\n"),
 	       semdata->sem_perm.mode, semdata->sem_perm.mode & 0777);
-- 
2.9.3

